client/packages/programs/src/JsonForms/components/ProgramEvent.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import { ControlProps, rankWith, uiTypeIs } from '@jsonforms/core';
+import { withJsonFormsControlProps } from '@jsonforms/react';
+import {
+  DetailInputWithLabelRow,
+  NumericTextInput,
+} from '@openmsupply-client/common';
+import {
+  DefaultFormRowSpacing,
+  DefaultFormRowSx,
+  FORM_LABEL_WIDTH,
+  useZodOptionsValidation,
+} from '../common';
+import {
+  useProgramEnrolments,
+  useProgramEvents,
+} from '../../api';
+
+import { z } from 'zod';
+
+export const programEventTester = rankWith(10, uiTypeIs('ProgramEvent'));
+
+const Options = z
+  .object({
+    documentType: z.string().optional(),
+    eventType: z.string(),
+    multiline: z.boolean().optional(),
+    rows: z.number().optional(),
+    /**
+     * Display option based on type.
+     */
+    display: z
+      .discriminatedUnion('type', [
+        z.object({ type: z.literal('number') }),
+        z.object({
+          type: z.literal('string'),
+          show: z.array(
+            z
+              .tuple([z.string(), z.string().optional()])
+              .rest(z.string().optional())
+          ),
+        }),
+      ])
+      .optional(),
+  })
+  .strict();
+type Options = z.infer<typeof Options>;
+
+const getDisplayOptions = (
+  data: string | null | undefined,
+  options?: Options
+) => {
+  let show =
+    options?.display?.type === 'string' ? options?.display?.show : null;
+
+  if (!show) {
+    return data;
+  }
+
+  let displayValue = show.find(value => value[0] === data)?.[1];
+  return displayValue ?? '';
+};
+
+const UIComponent = (props: ControlProps) => {
+  const { label, uischema, data } = props;
+
+  const { errors, options } = useZodOptionsValidation(
+    Options,
+    uischema.options
+  );
+
+  const { data: programEnrolment } =
+  useProgramEnrolments.document.programEnrolments({
+      filterBy: {
+          programEnrolmentId: {
+              equalTo: data.programEnrolmentId,
+            },
+        },
+    });
+    
+  const { data: events } = useProgramEvents.document.list(
+    {
+      patientId: programEnrolment?.nodes[0]?.patientId ?? '',
```

Comment:
This doesn't picks the correct patient id, e.g. when doing the testing for Andy and it will pick Tina here. This is because I haven't had an programEnrolmentId set for Andy . You could get the current document by name to get the correct `patientId` (documentName is in the config object in the props)

@CarlosNZ didn't you had a nicer way to get the current patient data?

Reply:
[889af6d](https://github.com/openmsupply/open-msupply/pull/1466/commits/889af6d6e4a13770388ab16c94e7018d3f659fdd)